apps/ledger-live-desktop/src/renderer/screens/platform/App.jsx

```diff
@@ -20,17 +20,32 @@ type Props = {
     url: string,
   },
   appId?: string,
+  location: {
+    hash: string,
+    params: {
+      [key: string]: string,
+    },
+    pathname: string,
+    search: string,
+  },
 };
 
-export default function PlatformApp({ match, appId: propsAppId }: Props) {
+export default function PlatformApp({ match, appId: propsAppId, location }: Props) {
   const history = useHistory();
-  const { state: urlParams, search } = useLocation();
+  const { params: urlParams, search, pathname } = location;
   const appId = propsAppId || match.params?.appId;
+  const sufix = pathname.substr(
+    pathname.indexOf(match.params.appId) + match.params.appId.length + 1,
+  );
 
   const localManifest = useLocalLiveAppManifest(appId);
   const remoteManifest = useRemoteLiveAppManifest(appId);
-
-  const manifest = localManifest || remoteManifest;
+  const manifest = Object.assign({}, localManifest || remoteManifest);
+  if (sufix !== "") {
+    const dappUrl = manifest?.params?.dappUrl?.split("?");
+    manifest.params.dappUrl =
```

Comment:
I think it would be worthwhile to write some unit tests for this as there's now quite a bit of complexity in this component. Maybe we can extract all of this to a hook called something like `useGetManifest(location, appId)` and test that?